lib/screens/v2/screen_data.ex

```diff
@@ -410,7 +410,14 @@ defmodule Screens.V2.ScreenData do
 
     instance_map
     |> Enum.group_by(
-      fn {paged_slot_id, _} -> Template.get_page(paged_slot_id) end,
+      fn
+        {paged_slot_id, instance} ->
+          if Map.has_key?(instance, :is_full_screen) and instance.is_full_screen do
```

Comment:
Separately: it feels a little janky to distinguish paged stuff on the left screen from that in the right screen flex-zone, by a key that happens to be on the one type of widget that can currently occupy that space.

Is there any other way to distinguish them?

If I'm following the code correctly, `paged_slot_id` should look something like `{0, :paged_main_content_left}` for a widget occupying one of the paged left screen slots. Checking for that slot ID would be a bit better, since it's independent of whatever widget is going in the slot.

So maybe something like:
```ex
|> Enum.group_by(
  fn
    {{_n, :paged_main_content_left} -> :paged_main_content_left
    {paged_slot_id, _} -> Template.get_page(paged_slot_id)
  end,
  fn {paged_slot_id, instance} -> {Template.unpage(paged_slot_id), instance} end
)
```

This still isn't ideal since it's specific to pre-fares, but maybe a bit better? :shrug:

Reply:
Yeah this is a tricky situation and definitely the what I spent the most time figuring out. This problem only exists because we get data for both screens at the same time (or if we ever have two flexzones on one screen for other apps). Because of this, I don't see any screen besides PreFare having this issue unless we add a flexzone to another screen.

Would it be cleaner to leverage `app_id` in this? Maybe bring that in as a parameter, refactor the key grouping anon function into a named function, and use a guard. What do you think?

Reply:
I think yeah, it might be best to just have special behavior for the pre-fare app. A separate named function leaves room for defining clauses for for other apps later, if need be, and separates the app-specific stuff from the common logic a little bit better.

Reply:
tl;dr Yes, a named `paged_content_key` (or some better name) function that also takes app ID as an argument, assuming that's easy to access from this function, sounds good.

Reply:
Got that added. Mind taking one more look?